### DISCUSSION TITLE
import umap from Seurat

### DISCUSSION THREAD

**Discussion by yamihn** [GreenleafLab/ArchR#1563] [Questions / Documentation]
hi, I have a question related to issue #455 is it possible to import umap embeddings produced in Seurat to archr? if I use this code:

seurat_coord= as.data.frame(seurat[["umap_wnn"]]@cell.embeddings)
umap <- data.frame(row.names= colnames(seurat), "seuratUMAP1" = seurat_coord$wnnUMAP_1, "seuratUMAP2" =  seurat_coord$wnnUMAP_2, check.names = FALSE)
hope_sub@embeddings$umap_wnn <- SimpleList(df = umap, params = list())

 plotEmbedding(ArchRProj = hope_sub, colorBy = "predictedGroup", size=1.2, name = "Sample", embedding = "umap_wnn")
Getting ImputeWeights
No imputeWeights found, returning NULL
ArchR logging to : ArchRLogs/ArchR-plotEmbedding-79621bf815e3-Date-2022-08-14_Time-00-02-23.log
If there is an issue, please report to github with logFile!
Errore in stringr::str_split(colnames(df)[1], pattern = "#", simplify = TRUE)[,  : 
  subscript out of bound. 
I don't know how to act exactly, could you show me how to proceed? thank you

[ArchR-plotEmbedding-79621bf815e3-Date-2022-08-14_Time-00-02-23.log](https://github.com/GreenleafLab/ArchR/files/9332238/ArchR-plotEmbedding-79621bf815e3-Date-2022-08-14_Time-00-02-23.log)

**Comment by rcorces** [comment]
as in the example in #455, I believe you need the `#` in the UMAP name. i.e. in your case `"seurat#UMAP1"` and `"seurat#UMAP2"` when you create the data.frame.

**Reply by mitchTwoTimes** [reply]
Hey @yamihn,

Here's how I did it
```
seurat.umap <- seurat.rna@reductions$global_umap

df <- DataFrame(row.names=proj$cellNames,
                "seurat#UMAP1" = seurat.umap@cell.embeddings[proj$cellNames, "UMAP_1"],
                "seurat#UMAP2" =  seurat.umap@cell.embeddings[proj$cellNames, "UMAP_2"],
                check.names = FALSE)
df$bc <- rownames(df)

atac.ph <- proj@cellColData
atac.ph$bc <- rownames(atac.ph)

ordered.coords <- merge(atac.ph, df, by = "bc")[, c("seurat#UMAP1", "seurat#UMAP2", "bc")]
rownames(ordered.coords) <- ordered.coords$bc
ordered.coords$bc <- NULL

proj@embeddings$UMAP_rna <- SimpleList(df = df, params = list())

```

**Reply by yamihn** [reply]
thank you!
